colin/utils/cont.py

```diff
@@ -0,0 +1,285 @@
+"""
+This is a temporary module to support unpriv way of interacting with container images.
+
+It will be migrated to conu sooner or later.
+"""
+import json
+import os
+import logging
+import shutil
+import subprocess
+from tempfile import mkdtemp
+
+from colin.core.exceptions import ColinException
+
+
+logger = logging.getLogger(__name__)
+
+
+def extract_file_from_tarball(tarball_path, file_path, wd):
+    """
+    Extract selected file (file_path) from tarball (tarball_path) within a selected
+    directory (wd).
+    """
+    tar_cmd = ["tar", "-xf", tarball_path, file_path]
+    out = subprocess.check_output(tar_cmd, cwd=wd)
+    if out:
+        logger.debug("output of tar command:")
+        logger.debug(out)
+
+
+class Image(object):
+    """ Image representation using skopeo, ostree and atomic tool """
+
+    def __init__(self, image_name, pull):
+        """
+        :param image_name: str, name of the image to access
+        :param pull: bool, pull the image from registry or from local dockerd
+        """
+        self.image_name = image_name
+        self.pull = pull
+        self._tmpdir = None
+        self._mount_point = None
+        self._ostree_path = None
+        self._layers_path = None
+
+        self.metadata = None
+        self._labels = None
+
+        self._pull_image()
+
+
+    @property
+    def tmpdir(self):
+        """ Temporary directory holding all the runtime data. """
+        if self._tmpdir is None:
+            self._tmpdir = mkdtemp(prefix="colin-", dir="/var/tmp")
```

Comment:
Would it be possible to generate tmp dir by NamesTemporaryDirectory?

Reply:
I moved it to cont.py